server/controllers/clubProfileController.js

```diff
@@ -6,15 +9,30 @@ module.exports = {
     },
     getById: function(req, res) {
         // TODO; req.params.id
+        clubProfile.findById(req.params.id)
+                .then(clubprofile => res.json(clubprofile))
+                .catch(err => res.status(422).json(err));
+
     },
     create: function(req, res) {
         // TODO; req.body contains profile information
+        clubProfile.create(req.body)
+                .then(newclubProfile => res.json(newclubProfile))
+                .catch(err => res.status(422).json(err))
     },
     update: function(req, res) {
         // TODO; req.body contains profile information and req.params.id
```

Comment:
have you confirmed the mongoose interface definition of `findOneAndUpdate` and `findById` and made sure that they are used this way? has any testing been performed?

Reply:
I've confirmed the functionality of each function and made the usage consistent throughout.